flocker/testtools/__init__.py

```diff
@@ -299,6 +302,28 @@ def loop(failure):
     return d
 
 
+def retry_effect_with_timeout(effect, timeout):
+    """
+    If ``effect`` fails, retry it until ``timeout`` expires.
+
+    :param Effect effect: The Effect to retry.
+    :param int timeout: Timeout in seconds.
```

Comment:
Please document the return value, e.g. `:return: An Effect that does what ``effect`` does, but retrying.`